fdbserver/storageserver.actor.cpp

```diff
@@ -6038,10 +6053,12 @@ ACTOR Future<Void> auditStorageQ(StorageServer* data, AuditStorageRequest req) {
 		    .detail("Range", req.range)
 		    .detail("AuditType", req.type)
 		    .detail("TargetServers", describe(req.targetServers));
-		if (e.code() != error_code_audit_storage_error) {
-			req.reply.sendError(audit_storage_failed());
-		} else {
+		if (e.code() == error_code_audit_storage_cancelled) {
+			req.reply.sendError(audit_storage_cancelled());
+		} else if (e.code() == error_code_audit_storage_error) {
 			req.reply.sendError(audit_storage_error());
+		} else {
+			req.reply.sendError(audit_storage_failed());
 		}
 		if (e.code() == error_code_actor_cancelled) {
```

Comment:
Usually this is the first thing to check in the `catch` block. Because if the actor is cancelled, some fields may not be valid any more. In this case, `data` could already be destroyed.